tests/test-remotes-config-dir.js

```diff
@@ -112,3 +112,42 @@ try {
 }
 
 print("ok config-remote-in-config-dir-fails");
+
+// Merging options for non-existent remote should fail
+try {
+    if (repo.remote_change(null, OSTree.RepoRemoteChange.MERGE,
+                           'nonexistent', 'http://nonexistent',
+                           null, null))
+        throw new Error('merge of options to missing remote should fail');
+} catch (e) {
+    if (!(e.matches(Gio.IOErrorEnum, Gio.IOErrorEnum.NOT_FOUND)))
+        throw e;
+}
+
+print("ok merge-missing-remote-fails");
+
+// Test merging options to remote in config dir
+let opts = GLib.Variant.new('a{sv}', {
+    'branches': GLib.Variant.new('as', ['master'])
+});
+repo.remote_change(null, OSTree.RepoRemoteChange.MERGE, 'baz',
+                   'http://baz2', opts, null);
```

Comment:
One issue this raises is that `* @url: URL for remote` is not currently `(nullable)`.  We could make it so...it'd be one of those subtle IABI breaks that only affects compiled languages.

(The reason to make it nullable is to make it more convenient to change the options without changing the URL)

Reply:
I hadn't thought of that. The `url` parameter is pretty odd in this operation. I had some more thoughts about this that I'll follow up on in the general PR conversation.